Sources/DarkModeCore/NSObject+DarkModeKit.m

```diff
@@ -7,23 +7,51 @@
 
 @import ObjectiveC;
 
-@implementation NSObject (DarkModeKit)
+@interface IMPContainer ()
 
-+ (BOOL)dm_swizzleInstanceMethod:(SEL)selector1 to:(SEL)selector2 {
-  Method method1 = class_getInstanceMethod(self, selector1);
-  Method method2 = class_getInstanceMethod(self, selector2);
+@property (readwrite) IMP imp; // Redefine as readwrite so we can set it in this file
 
-  if (!method1 || !method2) {
-    return NO;
-  }
+@end
+
+@implementation IMPContainer
 
-  if (class_addMethod(self, selector1, method_getImplementation(method2), method_getTypeEncoding(method2))) {
-    class_replaceMethod(self, selector2, method_getImplementation(method1), method_getTypeEncoding(method1));
-  } else {
-    method_exchangeImplementations(method1, method2);
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _imp = NULL;
   }
+  return self;
+}
+
+@end
+
+OBJC_EXPORT id objc_msgSendSuper2(struct objc_super *super, SEL op, ...);
+
+@implementation NSObject (DarkModeKit)
+
++ (BOOL)dm_swizzleSelector:(SEL)selector withBlock:(id (^)(IMPContainer *))implementationCreationBlock {
+  IMPContainer *container = [IMPContainer new];
```

Comment:
Why we need this container? Could we pass `IMP` directly?

Reply:
IMP is a C pointer and the swizzling block won't be able to catch it by reference